neutral_generation/smart_convert.py

```diff
@@ -91,16 +105,53 @@ def regex_token_replace(sentence: str, token: str, replacement: str) -> str:
     return sentence
 
 
-def score(sentence: str) -> float:
+def score(sentence: str, stride: int) -> float:
     """
     score the perplexity of a sentence
     :param sentence: input sentence
+    :param stride: calculate perplexity for every {stride} tokens, can trade-off speed for accuracy
```

Comment:
It's usually a good idea to include a comment when you replace a simple implementation with a more complex one why the change was made. This will give well-meaning later developers the info they need to not go ahead and revert to a simpler alternative.

Reply:
Ah, I see. Thanks for letting me know! I'll add in some more context around this change.